source/MRMesh/MRFilterCreaseEdges.cpp

```diff
@@ -29,67 +34,68 @@ namespace MR
             if ( componentLength < critLength )
                 creaseEdges -= selectedComponent;
         }
+    }
 
-        if ( !filterBranches )
-            return;
+    if ( !filterBranches )
+        return;
 
-        auto incidentVertices = getIncidentVerts( mesh.topology, creaseEdges );
+    auto incidentVertices = getIncidentVerts( mesh.topology, creaseEdges );
 
-        for ( auto v : incidentVertices )
+    for ( auto v : incidentVertices )
+    {
+        std::vector<std::pair<UndirectedEdgeId, VertId>> connections;
+        for ( UndirectedEdgeId ue : orgRing( mesh.topology, v ) )
         {
-            std::vector<std::pair<UndirectedEdgeId, VertId>> connections;
-            for ( UndirectedEdgeId ue : orgRing( mesh.topology, v ) )
-            {
-                if ( creaseEdges.test( ue ) )
-                    connections.push_back( { ue, mesh.topology.dest( ue ) } );
-            }
-
-            if ( connections.size() != 1 )
-                continue;
-
-            VertId prevVert = v;
-            VertId nextVert = connections[0].second;
-            float branchLength = 0;
-            UndirectedEdgeId ueCur = connections[0].first;
-            std::vector<UndirectedEdgeId> branch;
+            if ( creaseEdges.test( ue ) )
+                connections.push_back( { ue, mesh.topology.dest( ue ) } );
+        }
 
-            while ( true )
-            {
-                branch.push_back( ueCur );
-                branchLength += ( mesh.points[prevVert] - mesh.points[nextVert] ).length();
-                if ( branchLength > critLength )
-                    break;
+        if ( connections.size() != 1 )
+            continue;
 
-                connections.clear();
-                for ( EdgeId e : orgRing( mesh.topology, nextVert ) )
-                {
-                    if ( creaseEdges.test( e.undirected() ) )
-                        connections.push_back( { e.undirected(), mesh.topology.dest( e ) } );
-                }
+        VertId prevVert = v;
+        VertId nextVert = connections[0].second;
+        float branchLength = 0;
+        UndirectedEdgeId ueCur = connections[0].first;
+        std::vector<UndirectedEdgeId> branch;
```

Comment:
mb put branch declaratino out of loop, and just clear here, not to reallocate each iteration